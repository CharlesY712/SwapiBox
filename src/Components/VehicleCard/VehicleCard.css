.card {
  display: inline-block;
  margin: 10px;
  color: #feda4a;
}