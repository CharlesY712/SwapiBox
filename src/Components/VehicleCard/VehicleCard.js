import React from 'react';
import './VehicleCard.css';
import PropTypes from 'prop-types';

const VehicleCard = ({name, model, type, passengers}) => {
  return (
    <div className='vehicle-card'>
      <h1 className="vehicle-card-title">{name}</h1>
      <button className="fav-button" 
        onClick={() => console.log('clicked')}></button>
      <h4>Model: {model}</h4>
      <h4>Class: {type}</h4>
      <h4>Passengers: {passengers}</h4>
    </div>
  );
};

VehicleCard.propTypes = {
  name: PropTypes.string,
  model: PropTypes.string,
  type: PropTypes.string,
  passengers: PropTypes.string
};

export default VehicleCard;