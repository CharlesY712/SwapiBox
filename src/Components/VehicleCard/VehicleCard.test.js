import React from 'react';
import {shallow} from 'enzyme';
import VehicleCard from './VehicleCard';
import MockVehicleData from '../../MockData/MockVehicleData';

describe('VehicleCard', () => {
  const mockVehicleData = MockVehicleData;
  let wrapper = shallow(<VehicleCard 
    name={mockVehicleData.name}
    model={mockVehicleData.model}
    type={mockVehicleData.class}
    passengers={mockVehicleData.passengers}/>);
  
  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
