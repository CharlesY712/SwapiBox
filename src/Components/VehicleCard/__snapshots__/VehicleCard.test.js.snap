// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VehicleCard
    model="Digger Crawler"
    name="Sand Crawler"
    passengers="30"
    type={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="vehicle-card-title"
>
          Sand Crawler
</h1>,
        <button
          className="fav-button"
          onClick={[Function]}
/>,
        <h4>
          Model: 
          Digger Crawler
</h4>,
        <h4>
          Class: 
</h4>,
        <h4>
          Passengers: 
          30
</h4>,
      ],
      "className": "vehicle-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sand Crawler",
          "className": "vehicle-card-title",
        },
        "ref": null,
        "rendered": "Sand Crawler",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fav-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Model: ",
            "Digger Crawler",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Model: ",
          "Digger Crawler",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Class: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Class: ",
          undefined,
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Passengers: ",
            "30",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Passengers: ",
          "30",
        ],
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="vehicle-card-title"
>
            Sand Crawler
</h1>,
          <button
            className="fav-button"
            onClick={[Function]}
/>,
          <h4>
            Model: 
            Digger Crawler
</h4>,
          <h4>
            Class: 
</h4>,
          <h4>
            Passengers: 
            30
</h4>,
        ],
        "className": "vehicle-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sand Crawler",
            "className": "vehicle-card-title",
          },
          "ref": null,
          "rendered": "Sand Crawler",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fav-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Model: ",
              "Digger Crawler",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Model: ",
            "Digger Crawler",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Class: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Class: ",
            undefined,
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Passengers: ",
              "30",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Passengers: ",
            "30",
          ],
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
